src/utilities/HYPRE_utilities.h

```diff
@@ -83,9 +83,44 @@ typedef double HYPRE_Real;
 #endif
 
 #if defined(HYPRE_COMPLEX)
+/* support for float double and long double complex types */
+#if defined(HYPRE_SINGLE)
+typedef float _Complex HYPRE_Complex;
+#define HYPRE_MPI_COMPLEX MPI_C_FLOAT_COMPLEX
+#ifdef HYPRE_SEQUENTIAL
+#define MPI_C_FLOAT_COMPLEX hypre_MPI_COMPLEX
+#endif
+/* wrappers to some general complex functions */
+#define hypre_complex_abs cabsf
```

Comment:
Since we are already using function calls defined in `complex.c` for these, we should put the `#if defined` details inside the functions.  We definitely don't want lower-case `hypre_` stuff in this file.

Reply:
Thanks